### PR TITLE
ApiWorkflowClient: Include auto_tasks in selection_config_from_dict

### PR DESCRIPTION
**Pull request by MalteEbner**
## Description

Includes `auto_tasks` in the `selection_config_from_dict()` function so that passing them as dict is very simple.

## Tests

Adapted and new unittests

### CODE REVIEW

File: lightly/api/api_workflow_compute_worker.py [hunk 1]
```diff
@@ -624,16 +626,36 @@ def get_compute_worker_run_tags(self, run_id: str) -> List[TagData]:
 
 def selection_config_from_dict(cfg: Dict[str, Any]) -> SelectionConfigV4:
     """Recursively converts selection config from dict to a SelectionConfigV4 instance."""
-    strategies = []
-    for entry in cfg.get("strategies", []):
-        new_entry = copy.deepcopy(entry)
-        new_entry["input"] = SelectionConfigV4EntryInput(**entry["input"])
-        new_entry["strategy"] = SelectionConfigV4EntryStrategy(**entry["strategy"])
```
Comment (MalteEbner) [inline lightly/api/api_workflow_compute_worker.py:631]: Moved into the extra function `selection_config_entry_from_dict`

File: lightly/api/api_workflow_compute_worker.py [hunk 2]
```diff
@@ -624,16 +626,36 @@ def get_compute_worker_run_tags(self, run_id: str) -> List[TagData]:
...
 
+def selection_config_entry_from_dict(entry: Dict[str, Any]) -> AutoTask:
+    new_entry = copy.deepcopy(entry)
+    new_entry["input"] = SelectionConfigV4EntryInput(**new_entry["input"])
+    new_entry["strategy"] = SelectionConfigV4EntryStrategy(**new_entry["strategy"])
+    return SelectionConfigV4Entry(**new_entry)
+
+def auto_task_from_dict(entry: Dict[str, Any]) -> AutoTask:
+    auto_task_type_to_class =  {
+        "TILING": AutoTaskTiling,
```
Comment (MalteEbner) [inline lightly/api/api_workflow_compute_worker.py:648]: Other types are to be added here later.

File: tests/api_workflow/test_api_workflow_compute_worker.py [hunk 3]
```diff
@@ -325,7 +346,7 @@ def test_selection_config_from_dict__extra_key() -> None:
         api_workflow_compute_worker.selection_config_from_dict(cfg)
 
 
-def test_selection_config_from_dict__extra_stratey_key() -> None:
+def test_selection_config_from_dict__extra_strategy_key() -> None:
```
Comment (MalteEbner) [inline tests/api_workflow/test_api_workflow_compute_worker.py:349]: unrelated typo fix

File: lightly/api/api_workflow_compute_worker.py [hunk 4]
```diff
@@ -624,17 +626,39 @@ def get_compute_worker_run_tags(self, run_id: str) -> List[TagData]:
...
+        auto_tasks.append(auto_task_from_dict(entry=entry))
+    new_cfg["auto_tasks"] = auto_tasks
     return SelectionConfigV4(**new_cfg)
 
 
+def selection_config_entry_from_dict(entry: Dict[str, Any]) -> AutoTask:
+    new_entry = copy.deepcopy(entry)
+    new_entry["input"] = SelectionConfigV4EntryInput(**new_entry["input"])
+    new_entry["strategy"] = SelectionConfigV4EntryStrategy(**new_entry["strategy"])
+    return SelectionConfigV4Entry(**new_entry)
```
Comment (MalteEbner) [inline lightly/api/api_workflow_compute_worker.py:645]: copied from https://github.com/lightly-ai/lightly-core/blob/d70cca7a7cfa599ec95025bd9dcdb94b4cd7479e/onprem-docker/lightly_worker/src/lightly_worker/swagger_utils.py#L27-L31

**Review by guarin — APPROVED** [review]
